nautobot/dcim/filters/__init__.py

```diff
@@ -1524,3 +1535,56 @@ class DeviceRedundancyGroupFilterSet(NautobotFilterSet, StatusModelFilterSetMixi
     class Meta:
         model = DeviceRedundancyGroup
         fields = ["id", "name", "failover_strategy", "tags"]
+
+
+class InterfaceRedundancyGroupFilterSet(BaseFilterSet, NameSearchFilterSet):
+    """Filter for InterfaceRedundancyGroup."""
+
+    q = SearchFilter(filter_predicates={"name": "icontains"})
+    secrets_group = NaturalKeyOrPKMultipleChoiceFilter(
+        field_name="secrets_group",
+        queryset=SecretsGroup.objects.all(),
+        to_field_name="name",
+        label="Secrets group",
+    )
+    virtual_ip = MultiValueCharFilter(
+        method="filter_virtual_ip",
+        label="Virtual IP Address (address or ID)",
+    )
+
+    class Meta:
+        """Meta attributes for filter."""
+
+        model = InterfaceRedundancyGroup
+        fields = ["id", "name", "description", "secrets_group", "virtual_ip", "protocol", "protocol_group_id", "tags"]
+
+    # 2.0 TODO(jathan): Eliminate these methods.
```

Comment:
Do we know what's needed here and/or have an issue open for this?

Reply:
I asked Jathan, seems like filter methods in general are not a good pattern. In this case, we are trying to emulate 
`NaturalKeyOrPKMultipleChoiceFilter`, but `address` is not a model field for IP so we have to resort to the methods. It is also a pattern we have on `filter_primary_ip4/6` on `VirtualMachineFilterSet`

Reply:
Blocked by https://github.com/nautobot/nautobot/issues/4229